AuditHistoryExtractor/MyPluginControl.cs

```diff
@@ -427,29 +417,41 @@ private void RetrieveAuditHistory(RetrieveAuditHistoryMode retrieveMode)
                 {
                     try
                     {
+                        recordsExtracted = new List<Entity>();
+                        bool moreRecords = false;
+                        int page = 1;
+                        string cookie = string.Empty;
                         string newFetch = FetchXMLHelper.AddAttributeFilter(txtFetchXML.Text, cmbPrimaryKeySelectedItem.Value);
 
-                        fetchExpressionRecordsToExtract = new FetchExpression(newFetch);
-                        recordsExtracted = Service.RetrieveMultiple(fetchExpressionRecordsToExtract);
-
-                        if (recordsExtracted != null && !recordsExtracted.EntityName.Equals(currentEntitySelected))
-                        {
-                            ev.Result = string.Format(MessageMismatchEntitySelectedAndFetchXML, recordsExtracted.EntityName, currentEntitySelected);
-                            return;
-                        }
-                        if (recordsExtracted.Entities.Count == 0)
-                        {
-                            ev.Result = MessageNoAuditHistoryForSelectedRecords;
-                            return;
-                        }
-                        if (recordsExtracted.Entities.Count == 5000)
+                        do
                         {
-                            ev.Result = MessageToMuchRecords;
-
-                            return;
-                        }
-
-
+                            var xml = FetchXMLHelper.AddCookie(newFetch, cookie, page);
+                            EntityCollection collection = Service.RetrieveMultiple(new FetchExpression(xml));
+
+                            if (recordsExtracted != null && !collection.EntityName.Equals(currentEntitySelected))
```

Comment:
The recordsExtracted != null does not serve a reason or?

Reply:
I think you are right. I haven't changed all the code. This lines probably can be removed. I fear more, that this project is not maintained anymore and that change will not make it into a new release. :-(

Reply:
This need to be changed to !List.Any() to make sure, we retrieved anything from the fetch.